flink-runtime/src/main/java/org/apache/flink/runtime/jobgraph/JobVertex.java

```diff
@@ -133,11 +142,15 @@ public JobVertex(String name, JobVertexID id) {
 	 * @param name The name of the new job vertex.
 	 * @param primaryId The id of the job vertex.
 	 * @param alternativeIds The alternative ids of the job vertex.
+	 * @param operatorIds The ids of all operators contained in this job vertex.
+	 * @param alternativeOperatorIds The alternative ids of all operators contained in this job vertex-
 	 */
-	public JobVertex(String name, JobVertexID primaryId, List<JobVertexID> alternativeIds) {
+	public JobVertex(String name, JobVertexID primaryId, List<OperatorID> alternativeIds, List<OperatorID> operatorIds, List<OperatorID> alternativeOperatorIds) {
```

Comment:
We don't know at the time job JobVertex generation whether we need them, so we have to provide them eagerly.